curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-22-names-scores.english.md

```diff
@@ -75,7 +75,27 @@ const names = ['MARY','PATRICIA','LINDA','BARBARA','ELIZABETH','JENNIFER','MARIA
 <section id='solution'>
 
 ```js
-// solution required
+function nameScoreCalc(word) {
+  let sum = 0;
+  let alphabets = ["", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];
+
+  for (let i = 0 ; i < word.length; i++)
+    sum += alphabets.indexOf(word[i].toLowerCase());
+  return sum;
+}
+
+function namesScores(arr) {
+  arr = arr.sort();
```

Comment:
```suggestion
  arr = [...arr].sort();
```
Since the next line modifies the array passed to `nameScores`, it's best to defensively copy it.